spec/routes/api/cli/help_spec.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+require_relative "spec_helper"
+
+RSpec.describe Clover, "cli help" do
+  it "shows help for top level command with no arguments" do
+    expect(cli(%w[help])).to eq <<~OUTPUT
+      Usage: ubi [options] [subcommand [subcommand-options] ...]
+
+      Options:
+              --version                    show program version
+              --help                       show program help
+              --confirm=confirmation       confirmation value (not for direct use)
+
+      Subcommands: help pg vm
+    OUTPUT
+  end
+
+  it "shows help for specific command if given" do
+    expect(cli(%w[help help])).to eq <<~OUTPUT
+      Usage: ubi help [options] [command [subcommand]]
+
+      Options:
+          -r, --recursive                  also show documentation for all subcommands of command
+          -u, --usage                      only show usage
+    OUTPUT
+  end
+
+  it "shows help for specific subcommand if given" do
+    expect(cli(%w[help vm ssh])).to eq <<~OUTPUT
+      Usage: ubi vm location/(vm-name|_vm-ubid) [options] ssh [ssh-options --] [remote-cmd [remote-cmd-arg ...]]
+    OUTPUT
+  end
+
+  it "shows only usage if the -u flag is given" do
+    expect(cli(%w[help -u help])).to eq <<~OUTPUT
+      Usage: ubi help [options] [command [subcommand]]
+    OUTPUT
+  end
+
+  it "shows help for all subcommands of command if -r is given" do
+    expect(cli(%w[help -r vm])).to include <<~OUTPUT
+      Usage: ubi vm location/vm_name create [options] public_key
+
+      Options:
+          -6, --ipv6-only                  do not enable IPv4
+    OUTPUT
+  end
+
+  it "shows usage for all subcommands of command if -ru is given" do
+    expect(cli(%w[help -ru vm])).to include <<~OUTPUT
+      Usage: ubi vm list [options]
+      Usage: ubi vm location/vm_name create [options] public_key
```

Comment:
It might be worth considering if this should be two separate `Usage: ...` lines or one `Usage: ` header, with two newline delineated instances (ie similar to options).

You could also consider whether individual options should be inline, or if they warrant having their own lines as well. Either could be fine, but consistency matters.

Reply:
You get this output for `-ru`, showing all usages for the command and all subcommands.  The example only tests two lines, but there will generally be more (up to 27 lines currently).  Since all lines start with `Usage: `, we could strip that out, but I don't think we should add a header, since the user is explicitly asking only for usages.  Let me know if you think it's better to strip the `Usage: ` out.

I think inlining the options into the usage would result in excessively long lines in some cases, so to be consistent, I think it's best never to inline them.

Reply:
Ah, yeah that makes sense. I was focused on the output as I saw it in the test (which seemed inconsistent) and not so much the call pattern that resulted in it.

Sorry for the ambiguity, when I mentioned inline individual options, I meant more that in cases where there is only a single option, it would look something like:
```
Option: --version version
```
as opposed to
```
Option:
  --version  version
```

So they would still appear as their own section, it was more a matter of a question like: is inline unique to Usage, or is it a generalized special case for things to display where there is only one (which would otherwise be newline delineated). Does that make more sense?

Reply:
The option formatting comes directly from ruby's optparse library.  We can probably override it, but I don't think we should without a good reason.  Single options being on their own line seems fine to me, and it is consistent.

The `help -u` option is special case, where instead of asking the optparse library to return a string version of the parser (`OptionParser#to_s`) , we only take the option parser's banner.